lib/date.c

```diff
@@ -588,5 +592,10 @@ time_t mutt_parse_date(const char *s, struct Tz *tz_out)
     tz_out->zoccident = zoccident;
   }
 
-  return (mutt_mktime(&tm, 0) + tz_offset);
+  time_t time = mutt_mktime(&tm, 0);
+  /* Check we haven't overflowed the time (on 32-bit arches) */
+  if ((time != TIME_T_MAX) && (time != TIME_T_MIN))
+    time += tz_offset;
```

Comment:
Could we possibly overflow after the addition? Just asking...

Reply:
Absolutely.  It's now a game of 'whack-a-mole'.

I think we're probably OK.
Once we have the email's date in the correct timezone, then we're unlikely to alter it.
(I haven't checked this is true).

I chose `TIME_T_MIN`, `TIME_T_MAX` to better represent the extreme dates.
The alternative is to represent all out-of-range dates as 1970-01-01.